### ISSUE TITLE
Feature Request: DeleteRange with open range

### ISSUE THREAD

**Issue by ronag** [facebook/rocksdb#9911]
Would be nice if `DeleteRange` added an overload where begin and/or end are optional which means that it's open ended. Similar to lower/upper bound in the iterator API. I guess open begin could be emulated with `Slice("")`. However, open end is a bit trickier...

**Comment by cbi42** [comment]
Yes it would be nice feature to have, but is complicated to implement. I think `Slice("")` may not always work if a custom comparator is provided. As a workaround for now, user can provide smallest key and largest key or use iterator and SeekToFirst/Last() to find out such keys before calling DeleteRange().

**Comment by socketpair** [comment]
It's very important for me. I need to create an SST-file which should have `DeleteRange(NULL, NULL)` as the first operation.

Ingesting such kind of SST means erasing everything in db and replacing with the content from the rest of the SST file.
